src/roles/dullahan.py

```diff
@@ -205,37 +190,47 @@ def on_role_assignment(evt, cli, var, gamemode, pl, restart):
 
 @event_listener("succubus_visit")
 def on_succubus_visit(evt, cli, var, nick, victim):
-    if victim in TARGETS and TARGETS[victim] & var.ROLES["succubus"]:
-        TARGETS.difference_update(var.ROLES["succubus"])
-        pm(cli, victim, messages["dullahan_no_kill_succubus"])
-    if KILLS.get(victim) in var.ROLES["succubus"]:
-        pm(cli, victim, messages["no_kill_succubus"].format(KILLS[victim]))
-        del KILLS[victim]
+    user = users._get(victim) # FIXME
+    if user in TARGETS:
+        succ_target = False
+        for target in set(TARGETS[user]):
+            if target.nick in var.ROLES["succubus"]:
+                TARGETS[user].remove(target)
+                succ_target = True
+        if succ_target:
+            pm(cli, victim, messages["dullahan_no_kill_succubus"])
+    if user in KILLS and KILLS[user].nick in var.ROLES["succubus"]:
+        pm(cli, victim, messages["no_kill_succubus"].format(KILLS[user]))
```

Comment:
My bad, I was assuming format called __str__ which would return the full hostmask (mostly because __format__ is defined in the base class, so it wasn't immediately obvious)